### DISCUSSION TITLE
Limit of abilities of Extensions?

### DISCUSSION THREAD

**Discussion by mcswaip1** [nextgenhealthcare/connect#5949] [Q&A]
This is probably a really silly question, but what are the limits of Mirth Extensions? I wanted to add some functionality to the JavaScript editor and thought of Extensions, but then was unsure if they extended to that level.
Thoughts?
Thank you

**Comment by jonbartels** [comment]
I think it depends on what you want to do. I think the code snippets are pluggable, you can add your own snippets to do common operations. Like if you start typing FileUtil and get a popup with documentation.

Direct changes to the editor are probably not pluggable, but I am not sure.

https://github.com/kpalang/mirth-plugin-guide is a good reference. You can also search the code base for references to the `om.mirth.connect.plugins` package and see what is already pluggable in Mirth. Much of what Mirth does is plugin based so you can learn from the existing code. 

You should also sign up for Slack (see project README) and join #plugin-development. The active plugin hackers hang out there.

There was a recent thread discussing the editor and adding code formatting to the JS Editor - https://mirthconnect.slack.com/archives/C01QLL13HNU/p1695896143241049
